hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -404,7 +405,15 @@ public void run(Packet packet) throws Exception {
         Connection connection = packet.getConn();
         Address caller = connection.getEndPoint();
         try {
-            Object object = nodeEngine.toObject(packet);
+            // the packet was sent from 3.12 if the 4_0 flag is missing
+            // 4.0.1 and 4.2 members do not set this flag
```

Comment:
Same question with 4.1.

Reply:
Same as other comment, I need to add it on master but this PR seems good to go.